data-prepper-plugins/rds-source/src/main/java/org/opensearch/dataprepper/plugins/source/rds/leader/LeaderScheduler.java

```diff
@@ -20,13 +28,89 @@ public class LeaderScheduler implements Runnable {
     private final EnhancedSourceCoordinator sourceCoordinator;
     private final RdsSourceConfig sourceConfig;
 
+    private LeaderPartition leaderPartition;
+
     public LeaderScheduler(final EnhancedSourceCoordinator sourceCoordinator, final RdsSourceConfig sourceConfig) {
         this.sourceCoordinator = sourceCoordinator;
         this.sourceConfig = sourceConfig;
     }
 
     @Override
     public void run() {
+        LOG.info("Starting Leader Scheduler for initialization.");
+
+        while (!Thread.currentThread().isInterrupted()) {
```

Comment:
We should add a `shutdown` method and sets a boolean value that we can use rather than rely on interrupting the thread.

Reply:
Added.